ethcore/src/state.rs

```diff
@@ -222,6 +222,22 @@ impl State {
 		let options = TransactOptions { tracing: tracing, vm_tracing: false, check_nonce: true };
 		let e = try!(Executive::new(self, env_info, engine, vm_factory).transact(t, options));
 
+		// dao attack soft fork
+		if engine.schedule(&env_info).block_dao_transactions {
+			// collect all the addresses which have changed.
+			let addresses = self.cache.borrow().deref().iter().map(|(addr, _)| addr.clone()).collect::<Vec<_>>();
+
+			for a in addresses.iter() {
+				if self.code(a).map(|c| c.sha3() == H256::from("7278d050619a624f84f51987149ddb439cdaadfba5966f7cfaea7ad44340a4ba")).unwrap_or(false) {
```

Comment:
Should the hash maybe be a list in the case where more hashes need to be added in the coming days/future?